src/Client.ts

```diff
@@ -0,0 +1,15 @@
+import { BaseShipment, Shipment } from "./Shipment";
+import { Order } from "./types";
+
+export class Client {
+  shipment: Shipment;
```

Comment:
as it is singleton, you should init it here 
shipment = new BaseShipment()

Please, pay attention, I don't pass order inside of constructor. 
Instead you'd better pass order in `ship` method. 

Ship method can look like this:

```
ship(order) {
    const shipper = this.getShipper(order.fromZipCode)
    const cost = shipper.getCost(order.weight)
    const id = this. getShipmentID()

    return shipper.ship() -> returns final string
}
```